packages/core/src/api/toDecimal.ts

```diff
@@ -49,23 +49,22 @@ function getDecimal<TAmount>(
   formatter: Formatter<TAmount>
 ) {
   const absoluteFn = absolute(calculator);
+  const equalFn = equal(calculator);
+  const lessThanFn = lessThan(calculator);
+  const zero = calculator.zero();
 
   return (units: readonly TAmount[], scale: TAmount) => {
-    return units
-      .map((unit, index) => {
-        const isFirst = index === 0;
-        const isLast = units.length - 1 === index;
-
-        const unitAsString = formatter.toString(
-          isFirst ? unit : absoluteFn(unit)
-        );
-
-        if (isLast) {
-          return unitAsString.padStart(formatter.toNumber(scale), '0');
-        }
-
-        return unitAsString;
-      })
-      .join('.');
+    const integer = formatter.toString(units[0]);
+    const fractional = formatter.toString(absoluteFn(units[1]));
+
+    const scaleNumber = formatter.toNumber(scale);
+    const decimal = `${integer}.${fractional.padStart(scaleNumber, '0')}`;
+
+    // A leading negative zero is a special case because the toString
+    // formatter (generally String) won't preserve its negative sign.
+    const leadsWithZero = equalFn(units[0], zero);
+    const isNegative = lessThanFn(units[1], zero);
+
+    return leadsWithZero && isNegative ? `-${decimal}` : decimal;
```

Comment:
```suggestion
    // A leading negative zero is a special case because the `toString`
    // formatter won't preserve its negative sign (since 0 === -0).
    return leadsWithZero && isNegative ? `-${decimal}` : decimal;
```